class.jetpack-client.php

```diff
@@ -285,4 +300,36 @@ static function wpcom_json_api_request_as_blog( $path, $version = self::WPCOM_JS
 		return Jetpack_Client::remote_request( $validated_args, $body );
 	}
 
+	/**
+	 * Takes an array or similar structure and recursively turns all values into strings. This is used to
+	 * make sure that body hashes are made ith the string version, which is what will be seen after a
+	 * server pulls up the data in the $_POST array.
+	 *
+	 * @param array|mixed $data
+	 *
+	 * @return array|string
+	 */
+	public static function _stringify_data( $data ) {
```

Comment:
As noted in the comment this data is sent through HTTP requests, this means that types are not maintained and the two sides may end up with different content. AKA, and integer on one side would be a string on the other. This would lead to body hash mismatches and serialize maintains the type. This converts all the array's values into bare strings, so that an encode of the data will have consistent data on either side.